api/agent/agent.go

```diff
@@ -542,6 +540,15 @@ func (s *hotSlot) exec(ctx context.Context, call *call) error {
 	case err := <-s.errC: // error from container
 		return err
 	case err := <-errApp: // from dispatch
+		if s.fatalErr == nil && err != nil {
+			if models.IsAPIError(err) {
+				s.fatalErr = err
+			} else if err == protocol.ErrExcessData {
+				s.fatalErr = err
+				// suppress excess data error, but do shutdown the container
+				return nil
+			}
+		}
```

Comment:
other future idea: the errors we expect we could make an error interface along the lines of `interface{ Fatal() bool }` and then a little `func IsFatal(err) bool { f, ok := err.(FatalErr); return ok && f.Fatal() }` -- this would allow us to limit to certain errors we expect like too much output, protocol error, size error, etc. more tersely. though from dispatch probably any error we get is fatal?

Reply:
yes, that sounds good. Dispatch() results are tricky... And this keeps changing based on policy discussions.. So far, three types we have encountered;

1) client messed up, timeout, etc, container goes on
2) protocol corrupt, client gets error, container terminates
3) protocol corrupt, but we could service *this* request, client gets OK, container terminates
4) protocol corrupt, client gets error, container goes on (we honor '2' instead)

I think we still have some to-do on case '1'.